### PR TITLE
fix(core): allow credentials to be called without token provider and no guest mode

### PR DESCRIPTION
**Pull request by elorzafe**
#### Description of changes

Disable check for `allowGuestAccess: true` in order to call `credentialsProvider` when no tokens are present

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/core/src/singleton/Auth/index.ts [hunk 1]
```diff
@@ -77,7 +68,7 @@ export class AuthClass {
 						forceRefresh: options.forceRefresh,
 					}
 				);
-		} else if (this.authConfig.Cognito.allowGuestAccess) {
+		} else {
```
Comment (Samaritan1011001) [inline packages/core/src/singleton/Auth/index.ts:71]: The allowGuestAccess is still being [used](https://github.com/Samaritan1011001/amplify-js/blob/e561195255d4eb0e92bb1f2a4d5c0fbd31b9c28b/packages/auth/src/providers/cognito/credentialsProvider/credentialsProvider.ts#L65) within the CognitoCredentialsProvider 

Are we removing this flag from the config? I would be concerned if so.

File: packages/auth/src/providers/cognito/apis/signInWithRedirect.ts [hunk 2]
```diff
@@ -40,7 +40,7 @@ export function signInWithRedirect(
 	const authConfig = Amplify.getConfig().Auth?.Cognito;
 	assertTokenProviderConfig(authConfig);
 	assertOAuthConfig(authConfig);
-
+	store.setAuthConfig(authConfig);
```
Comment (jimblanc) [inline packages/auth/src/providers/cognito/apis/signInWithRedirect.ts:43]: Readability nit: Would prefer to move `store` declaration ahead of `signInWithRedirect` to make this easier to ready

File: packages/auth/src/providers/cognito/apis/signInWithRedirect.ts [hunk 3]
```diff
@@ -347,45 +347,49 @@ function validateState(state?: string | null): asserts state {
 	}
 }
 
+async function parseRedirectURL() {
```
Comment (jimblanc) [inline packages/auth/src/providers/cognito/apis/signInWithRedirect.ts:350]: Should this change be included here? We'll have merge conflicts with your other PR. PR to pull that change into the correct branch: https://github.com/aws-amplify/amplify-js/pull/12017

Comment (elorzafe) [reply packages/auth/src/providers/cognito/apis/signInWithRedirect.ts:350]: I will revert that change

File: packages/core/src/singleton/Auth/index.ts (hunk 1 above)
Comment (elorzafe) [reply packages/core/src/singleton/Auth/index.ts:71]: Is validated on the provider, `fetchAuthSession` should ignore that setting

**Review by Samaritan1011001 — APPROVED** [review]

**Review by jimblanc — APPROVED** [review]

**Pull request merged**